src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4080,19 +4100,39 @@ EqualOpExpressionLists(List *firstOpExpressionList, List *secondOpExpressionList
  * WorkerLimitCount checks if the given extended node contains a limit node, and
  * if that node can be pushed down. For this, the function checks if this limit
  * count or a meaningful approximation of it can be pushed down to worker nodes.
- * If they can, the function returns the limit count. Otherwise, the function
+ * If they can, the function returns the limit count.
+ *
+ * The limit push-down decision tree is as follows:
+ *                                 group by?
+ *                              1/           \0
+ *                          order by?         (exact pd)
+ *                       1/           \0
+ *           has order by agg?           has aggregate
+ *            1/           \0            1/           \0
+ *     can approximate?    (exact pd)   (no pd)     (exact pd)
+ *      1/       \0
+ * (approx pd)   (no pd)
+ *
+ * When an offset is present, the offset value is added to limit because for a query
+ * with LIMIT x OFFSET y, (x+y) records should be pulled from the workers.
+ *
+ * If no limit or offset is present or can be pushed down, then WorkerLimitCount
  * returns null.
  */
 static Node *
 WorkerLimitCount(MultiExtendedOp *originalOpNode)
 {
-	Node *workerLimitCount = NULL;
+	Node *workerLimitNode = NULL;
 	List *groupClauseList = originalOpNode->groupClauseList;
 	List *sortClauseList = originalOpNode->sortClauseList;
 	List *targetList = originalOpNode->targetList;
+	bool hasOrderByAggregate = HasOrderByAggregate(sortClauseList, targetList);
+	bool hasAggregateTarget = contain_agg_clause((Node *) targetList);
+	bool canPushDownLimit = false;
+	bool canApproximate = false;
 
 	/* no limit node to push down */
-	if (originalOpNode->limitCount == NULL)
+	if (originalOpNode->limitCount == NULL && originalOpNode->limitOffset == NULL)
```

Comment:
Yes, it was a silly mistake 😅  assuming limit = 0 when limit is not present. Changed accordingly.